core/src/toga/widgets/splitcontainer.py

```diff
@@ -1,106 +1,118 @@
+              that widget in the split.
+            * A widget. The widget will be given a flex value of 1.
+
+        """
         super().__init__(id=id, style=style)
-        ######################################################################
-        # 2022-09: Backwards compatibility
-        ######################################################################
-        # factory no longer used
-        if factory:
-            warnings.warn("The factory argument is no longer used.", DeprecationWarning)
-        ######################################################################
-        # End backwards compatibility.
-        ######################################################################
-
-        self._direction = direction
+
         self._content = []
-        self._weight = []
 
         # Create a platform specific implementation of a SplitContainer
         self._impl = self.factory.SplitContainer(interface=self)
 
-        self.content = content
+        if content:
+            self.content = content
         self.direction = direction
 
     @property
-    def content(self):
-        """The sub layouts of the `SplitContainer`.
+    def enabled(self) -> bool:
+        """Is the widget currently enabled? i.e., can the user interact with the widget?
+
+        SplitContainer widgets cannot be disabled; this property will always return
+        True; any attempt to modify it will be ignored.
+        """
+        return True
 
-        Returns:
-            A ``list`` of :class:`toga.Widget`. Each element of the list
-            is a sub layout of the `SplitContainer`
+    @enabled.setter
+    def enabled(self, value):
+        pass
+
+    def focus(self):
+        "No-op; SplitContainer cannot accept input focus"
+        pass
+
+    @property
+    def content(self) -> list[Widget]:
+        """The widgets displayed in the SplitContainer.
+
+        When retrieved, only the list of widgets is returned.
+
+        When setting the value, the list must have at least 2 elements; each item in the
+        list can be either:
+
+        * A tuple containing of a widget and the initial flex value to apply to that
+          widget in the split.
+        * A widget. The widget will be given a flex value of 1.
 
-        Raises:
-            ValueError: If the list is less than two elements long.
         """
         return self._content
 
     @content.setter
     def content(self, content):
-        if content is None:
-            self._content = None
-            return
-
-        if len(content) < 2:
-            raise ValueError("SplitContainer content must have at least 2 elements")
-
-        self._content = []
-        for position, item in enumerate(content):
+        if content is None or len(content) != 2:
+            raise ValueError(
+                "SplitContainer content must be a list with exactly 2 elements"
+            )
```

Comment:
If the constructor accepts None, then the property should as well.

However, I think it would be better if we disallowed None in both places, and instead allowed each *element* of the pair to be None, meaning that side of the split is empty. Then the default constructor argument becomes `(None, None)`.

I also suggest making the property return a tuple rather than a list, to prevent the user from attempting to assign to its elements. When setting the value, it should still accept any sequence of length 2.

Reply:
Agreed it should be a tuple on return, accepting any 2-length sequence.

Also agreed the None case is a wart; I don't particularly like the way that None is handled here. 

Historically, the complication was Cocoa; it doesn't have an explicit "left/right"; it's a container of as many children as you want (this is why it used to allow N-way splits). If you don't specify 2 children, you don't get a splitter. However, it occurs to me that as a result of the base Container changes, it *will* have 2 children - they'll just be containers with no content. Which is exactly what the user asked for.

Testbed and core API have been updated to reflect this.